pkgs/test/nixpkgs-check-by-name/scripts/run-local.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2016
+
+set -euo pipefail
+
+trace() { echo >&2 "$@"; }
+
+cleanup_commands=()
+cleanup() {
+    trace -n "Cleaning up.. "
+    # Run all cleanup commands in inverse order
+    for (( i=${#cleanup_commands[@]}-1; i>=0; i-- )); do
+        eval "${cleanup_commands[i]}"
+    done
+    trace "Done"
+}
+trap cleanup exit
+
+tmp=$(mktemp -d)
+cleanup_commands+=('rmdir "$tmp"')
```

Comment:
The cleanup commands should generally be set up *before* the command which creates the resource, so that if the resource is created but the command returns a non-zero exit code for some other reason the cleanup still happens.

Reply:
Ah good point! Done in 206361ab3889aeaee4428d8cd2e175cbdc5fbdc7